src/main/java/br/unb/cic/analysis/svfa/confluence/DFPConfluenceAnalysis.java

```diff
@@ -61,16 +61,21 @@ private Set<ConfluenceConflict> intersectPathsByLastNode(Set<List<StatementNode>
         Set<ConfluenceConflict> result = new HashSet<>();
         for (List<StatementNode> path : paths2) {
             StatementNode lastNode = getLastNode(path);
-            for (StatementNode stm: pathEndHash.keySet()){
-                if (lastNode.equals(stm)){
-                    result.add(new ConfluenceConflict(pathEndHash.get(stm), path));
+            for (StatementNode stmt: pathEndHash.keySet()){
+                if (containsKey(lastNode, stmt)) {
+                    result.add(new ConfluenceConflict(pathEndHash.get(stmt), path));
                 }
```

Comment:
@galilasmb não era esse extract. era para tu olhar o código como era antes. estava melhor do que agora. containsKey, como o nome diz, envolve todo o loop. é só olhar o código anterior ao que fiz o primeiro comentário

Reply:
O outro código usava a função nativa containsKey, este não funciona porque tem que usar o equals alterado lá no StatementNode.

Reply:
O que podemos fazer é remover esse for daí, mas ficaria bem mais veio o código, porque teria que passar pathEndHash e as outras variáveis, além de result, que é alterado como variável local.

Reply:
a minha sugestão era aplicar extract method, levando o for interno para um metodo que verifica se containsKey do seu jeito

Reply:
Feito.